cmd/templ/lspcmd/proxy/server.go

```diff
@@ -344,21 +346,36 @@ func (p *Server) Completion(ctx context.Context, params *lsp.CompletionParams) (
 			if !ok {
 				continue
 			}
-			imp := addImport(doc.Lines, item.Detail)
-			te := lsp.TextEdit{
-				Range: lsp.Range{
-					Start: lsp.Position{Line: uint32(imp.LineIndex), Character: 0},
-					End:   lsp.Position{Line: uint32(imp.LineIndex), Character: 0},
-				},
-				NewText: imp.Text,
+
+			te := handleImportTextEdit(doc, &item, item.Detail)
+			item.AdditionalTextEdits = []lsp.TextEdit{te}
+		}
+		if strings.Contains(item.Detail, "(from \"") {
```

Comment:
Is there anything else we could do here around checking the completion kind? Just to limit the likelihood that this regexp matches with something else and triggers when it shouldn't?

Reply:
what we can do here is to replace the simple `strings.Contains` with another, more explicit, regex expression. Something like `^(type|func|const|var)?\s?\(from\s".+"\)$`. This will also avoid panics on `pkg := pkgFromImportDetail.FindStringSubmatch(item.Detail)[0]` because they complement each other

Reply:
check last commit